webserver/static/js/custom_blocks_events.js

```diff
@@ -134,10 +134,12 @@ Blockly.Python.event_button_object = function(block) {
 
     var eventName = block.getFieldValue('EventName');
     var funcName = Blockly.Python.variableDB_.getName(eventName, Blockly.Procedures.NAME_TYPE);
-    var x = Blockly.Python.valueToCode(block, 'X', Blockly.Python.ORDER_ATOMIC) || '0';
-    var y = Blockly.Python.valueToCode(block, 'Y', Blockly.Python.ORDER_ATOMIC) || '0';
-    var width = Blockly.Python.valueToCode(block, 'Width', Blockly.Python.ORDER_ATOMIC) || '1';
-    var height = Blockly.Python.valueToCode(block, 'Height', Blockly.Python.ORDER_ATOMIC) || '1';
+
+    let values = this.getInput('Values')
+    let x = values.fieldRow.find(({name}) => name === 'X').getValue() || 0
+    let y = values.fieldRow.find(({name}) => name === 'Y').getValue() || 0
+    let width  = values.fieldRow.find(({name}) => name === 'Width').getValue() || 1
+    let height = values.fieldRow.find(({name}) => name === 'Height').getValue() || 1
```

Comment:
We have nearly identical code above in callback for `setOnChange()`.
The only difference is that the latter is missing the default fallback values `|| 0` or `|| 1`. Shouldn't we always use fallback values?
Moreover, can we reduce duplicated code by extracting a function for getting these values?